code/modules/mob/living/simple_animal/hostile/tree.dm

```diff
@@ -27,13 +27,28 @@
 	emote_taunt = list("growls")
 	taunt_chance = 20
 
-	//Space carp aren't affected by atmos.
-	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	//Space carp aren't affected by atmos. //Space Carps no. But trees Yes
+	atmos_requirements = list("min_oxy" = 2, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	unsuitable_atmos_damage = 5
 	minbodytemp = 0
+	maxbodytemp = 400
 
 	faction = list("hostile")
 	var/drop_type = /obj/item/stack/sheet/mineral/wood
 
+/mob/living/simple_animal/hostile/tree/Life()
+	..()
+	//Thank'e Phil  for the simplification
+	if(istype(src.loc, /turf/simulated))
+		var/turf/simulated/T = src.loc
+		if(T.air)
+			var/co2 = T.air.carbon_dioxide
+			if(co2 > 0)
+				if(prob(25))
+					var/amt = min(co2, 9)
+					T.air.carbon_dioxide -= amt
+					T.atmos_spawn_air(SPAWN_OXYGEN, amt/3*2)

```

Comment:
Maths is maths @oisin100 I'm just saying, don't go too realistic with it, Most of the game is a simple
"I removed 10 Oxygen while breathing, I will expell 10 Co2"
so ideally these trees would be consistent with that:
"I removed 10 Co2 while Photosynthesising, I will expell 10 oxygen"